### PR TITLE
Move to logging

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This converts all print statements somewhat bluntly to logging.

This could probably be improved by also reworking the logging output to include things like which board/runner the info is coming from, though for now this is at least no worse than it was previously.

### CODE REVIEW

**Review by kierdavis — DISMISSED** [review]
This looks fine to me.

It would be good to make the use of format strings more consistent in a follow-up PR - currently it's using a mixture of old-style (`%s`) and new-style (`{}`) strings.

**Comment by PeterJCLaw** [comment]
This is deliberate -- `logging` does automatic conversion of `%` strings (for better or worse), so the only way to take advantage of the delayed formatting is to use that mechanism.
I agree it would be nice to be able to use `{}` style formatting, but I tend to prefer to keep the lazy evaluation when using logging.

File: robotd/devices.py [hunk 1]
```diff
@@ -299,7 +306,7 @@ def start(self):
         self._ultrasound_value = None
 
         self.make_safe()
-        print('Finished initialising servo assembly on {}'.format(device))
+        LOGGER.debug('Finished initialising servo assembly on {}'.format(device))
```
Comment (PeterJCLaw) [inline robotd/devices.py]: oops, will convert this to use `%` formatting for consistency.

File: robotd/master.py [hunk 2]
```diff
@@ -90,7 +93,7 @@ def _create_server_socket(self):
 
         self.socket_path.chmod(0o777)
 
-        print('Listening on:', self.socket_path)
+        LOGGER.info('Listening on:', self.socket_path)
```
Comment (PeterJCLaw) [inline robotd/master.py]: darn, these need a placeholder adding.

**Review by PeterJCLaw — COMMENTED** [review]
changes needed.

**Comment by kierdavis** [comment]
Fair enough - I didn't know that `logging` only supported `%`-style format strings for its lazy formatting.

File: robotd/devices.py (hunk 1 above)
Comment (PeterJCLaw) [reply robotd/devices.py]: e1d68d5

File: robotd/master.py (hunk 2 above)
Comment (PeterJCLaw) [reply robotd/master.py]: e65b535

**Review by kierdavis — APPROVED** [review]

**Review by Adimote — APPROVED** [review]

**Pull request merged**